### ISSUE TITLE
add support for embedded structs

### ISSUE THREAD

**Issue by taobaohi** [blockloop/scan#16]
type Struct1 struct {
	Name1 string `db:"nam_e1" ` // no val
       	Name2 string `db:"name2" `  // has val
}
type Struct2 struct {
       Struct1 
	Name3 string `db:"nam_e3" ` // has val
       	Name4 string `db:"name4" `  // has val
}
func main(){
    sql="select nam_e1,name2,nam_e3,name4"
   rows, err := conn.Query(sql)
   var st2 []Struct2 
   err = scan.Rows(&st2 , rows)
  //  Name1 string `db:"nam_e1" ` // no value
 // st2.nam_e1 no value
}

**Comment by blockloop** [comment]
Embedded structs are not implemented

**Comment by schmath** [comment]
```go
// fieldByName gets a struct's field by first looking up the db struct tag and falling
// back to the field's name in Title case.
func fieldByName(v reflect.Value, name string, strict bool) reflect.Value {
	if val := fieldByNameRecursion(v, name, strict); val != reflect.ValueOf(nil) {
		return val
	}
	if strict {
		return reflect.ValueOf(nil)
	}
	return v.FieldByName(strings.Title(name))
}

func fieldByNameRecursion(v reflect.Value, name string, strict bool) reflect.Value {
	typ := v.Type()
	val := reflect.ValueOf(nil)
	for i := 0; i < v.NumField(); i++ {
		field := typ.Field(i)
		tag, ok := field.Tag.Lookup("db")
		if ok && tag == name {
			return v.Field(i)
		}
		if field.Anonymous {
			val = fieldByName(reflect.New(field.Type).Elem(), name, strict)
			if val != reflect.ValueOf(nil) {
				return val
			}
		}
	}
	return val
}
```

**Comment by blockloop** [comment]
Fixed in #29
